File: test_backend.py
import requests
import json

API_URL = 'http://127.0.0.1:8000'

def test_backend():
    print('Testing Backend Connection with Python...')

    # 1. Health Check
    try:
        res = requests.get(f'{API_URL}/', timeout=5)
        print(f'✅ Health Check Passed: {res.json()}')
    except Exception as e:
        print(f'❌ Health Check Failed: {e}')

    # 2. Calculate Bill
    try:
        res = requests.post(f'{API_URL}/calculate-bill', json={'kwh': 300}, timeout=5)
        if res.status_code == 200:
            print(f'✅ Calculate Bill Passed: {res.json()}')
        else:
            print(f'❌ Calculate Bill Failed: {res.text}')
    except Exception as e:
        print(f'❌ Calculate Bill Error: {e}')

    # 3. Predict Appliance (AC)
    try:
        payload = {
            "appliance_name": "ac",
            "details": {
                "total_kwh_monthly": 300,
                "ac_hours_per_day": 6,
                "ac_tonnage": 1.5,
                "ac_star_rating": 3,
                "num_ac_units": 1,
                "ac_type": "split",
                "ac_usage_pattern": "moderate"
            },
            "total_bill": 300
        }
        res = requests.post(f'{API_URL}/predict-appliance', json=payload, timeout=5)
        if res.status_code == 200:
            print(f'✅ Predict Appliance (AC) Passed: {res.json()}')
        else:
            print(f'❌ Predict Appliance Failed: {res.text}')
    except Exception as e:
        print(f'❌ Predict Appliance Error: {e}')

if __name__ == '__main__':
    test_backend()

File: tests/test_backend.js
const axios = require('axios');

const API_URL = 'http://127.0.0.1:8000';

async function testBackend() {
    console.log('Testing Backend Connection...');

    // 1. Health Check
    try {
        const health = await axios.get(API_URL + '/');
        console.log('✅ Health Check Passed:', health.data);
    } catch (error) {
        console.error('❌ Health Check Failed:', error.message);
        return;
    }

    // 2. Calculate Bill
    try {
        const bill = await axios.post(API_URL + '/calculate-bill', { kwh: 300 });
        console.log('✅ Calculate Bill Passed:', bill.data);
    } catch (error) {
        console.error('❌ Calculate Bill Failed:', error.message);
        if (error.response) console.error('Response:', error.response.data);
    }

    // 3. Predict Appliance (AC)
    try {
        const payload = {
            appliance_name: "ac",
            details: {
                total_kwh_monthly: 300,
                ac_hours_per_day: 6,
                ac_tonnage: 1.5,
                ac_star_rating: 3,
                num_ac_units: 1,
                ac_type: "split",
                ac_usage_pattern: "moderate"
            },
            total_bill: 300
        };
        const prediction = await axios.post(API_URL + '/predict-appliance', payload);
        console.log('✅ Predict Appliance (AC) Passed:', prediction.data);
    } catch (error) {
        console.error('❌ Predict Appliance Failed:', error.message);
        if (error.response) console.error('Response:', error.response.data);
    }
}

testBackend();
